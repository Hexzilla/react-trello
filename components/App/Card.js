import React, { Component } from 'react'

export default class Card extends Component {
  render () {
    return <article className='card'>
      <header>{this.props.title}</header>
      <div className='detail'>{this.props.description}</div>
    </article>
  }
}

Card.propTypes = {
  title: React.PropTypes.string.isRequired,
  description: React.PropTypes.string.isRequired
}
