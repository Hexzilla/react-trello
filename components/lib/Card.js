import React, {Component} from 'react'

export default class Card extends Component {
  render() {
    const {title, description, rightHeader, ...otherProps} = this.props
    return <article className='card' key={title} {...otherProps}>
      <header>
        <span className='title'>{title}</span>
        <span className='rightContent'>{rightHeader}</span>
      </header>
      <div className='detail'>{description}</div>
    </article>
  }
}

Card.propTypes = {
  title: React.PropTypes.string.isRequired,
  description: React.PropTypes.string,
  rightHeader: React.PropTypes.string,
  onClick: React.PropTypes.func,
  id: React.PropTypes.string
}
