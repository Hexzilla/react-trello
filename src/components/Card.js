import React, {Component} from 'react'
import PropTypes from 'prop-types'

import {
  MovableCardWrapper,
  CardHeader,
  CardRightContent,
  CardTitle,
  Detail,
  Footer
} from 'rt/styles/Base'
import Tag from './Card/Tag'
import DeleteButton from 'rt/widgets/DeleteButton'

class Card extends Component {
  onDelete = e => {
    this.props.onDelete()
    e.stopPropagation()
  }

  render()  {
    const {
      showDeleteButton,
      style,
      tagStyle,
      onClick,
      onDelete,
      className,
      id,
      title,
      label,
      description,
      tags,
    } = this.props

    return (
      <MovableCardWrapper
        data-id={id}
        onClick={onClick}
        style={style}
        className={className}
      >
        <CardHeader>
          <CardTitle>{title}</CardTitle>
          <CardRightContent>{label}</CardRightContent>
          {showDeleteButton && <DeleteButton onClick={this.onDelete} />}
        </CardHeader>
        <Detail>{description}</Detail>
        {tags && tags.length> 0 && (
          <Footer>
            {tags.map(tag => (
              <Tag key={tag.title} {...tag} tagStyle={tagStyle} />
            ))}
          </Footer>
        )}
      </MovableCardWrapper>
      )
  }
}

Card.propTypes = {
  showDeleteButton: PropTypes.bool,
  onDelete: PropTypes.func,
  onClick: PropTypes.func,
  style: PropTypes.object,
  tagStyle: PropTypes.object,
  className: PropTypes.string,
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  label: PropTypes.string,
  description: PropTypes.string,
  tags: PropTypes.array,
}

Card.defaultProps = {
  showDeleteButton: true,
  onDelete: () => {},
  onClick: () => {},
  style: {},
  tagStyle: {},
  title: 'no title',
  description: '',
  label: '',
  tags: [],
  className: ''
}

export default Card
