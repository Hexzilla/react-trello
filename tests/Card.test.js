import React from 'react'
import renderer from 'react-test-renderer'
import Card from '../components/lib/Card'

describe('<Card/>', () => {
  it('snapshot test', () => {
    let spyClicked = false

    const component = renderer.create(
      <Card key='card1'
        title='Card title'
        description='New Description'
        rightHeader='Overdue'
        onClick={() => spyClicked = true} />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    tree.props.onClick()
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(spyClicked).toBe(true)
  })
})
