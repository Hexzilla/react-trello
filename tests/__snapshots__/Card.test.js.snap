exports[`<Card/> snapshot test 1`] = `
<article
  className="card drag-item"
  onClick={[Function]}>
  <header>
    <span
      className="title">
      Card title
    </span>
    <span
      className="rightContent">
      Overdue
    </span>
  </header>
  <div
    className="detail">
    New Description
  </div>
</article>
`;

exports[`<Card/> snapshot test 2`] = `
<article
  className="card drag-item"
  onClick={[Function]}>
  <header>
    <span
      className="title">
      Card title
    </span>
    <span
      className="rightContent">
      Overdue
    </span>
  </header>
  <div
    className="detail">
    New Description
  </div>
</article>
`;
